BepInEx.PatcherProvider/BepInExPatcherLoader.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System.IO;
+
+namespace BepInEx.PatcherProvider;
+
+public class BepInExPatcherLoader : IPluginLoader
+{
+    public string AssemblyIdentifier { get; internal set; }
+    public string AssemblyHash { get; internal set; }
+    private byte[] assemblyData;
```

Comment:
Holding the byte array forever is kind of suboptimal, especially when embedded resources are used. Maybe chainloader finishing could trigger all of these to get cleared? Or make this implement `IDisposable` and have chainloader dispose it after it will no longer need the ass data? This way would also force others to implement a Dispose method, forcing them to consider this issue I suppose.

Reply:
Added IDosposable requirement and implemented it on the default 2 providers